src/solution.c

```diff
@@ -420,16 +421,20 @@ static void solution_thread(void *arg)
     gps_time_t rec_time = rx2gpstime(rec_tc);
     u8 n_ready = 0;
     channel_measurement_t meas[MAX_CHANNELS];
+    SID_LIST_DECL(sids_in_track, MAX_CHANNELS);
     for (u8 i=0; i<nap_track_n_channels; i++) {
       tracking_channel_lock(i);
       if (use_tracking_channel(i)) {
         tracking_channel_measurement_get(i, rec_tc, &meas[n_ready]);
+        if (sid_list_find(&sids_in_track, meas[n_ready].sid) < 0) {
+          sid_list_add(&sids_in_track, meas[n_ready].sid);
+        }
         n_ready++;
       }
       tracking_channel_unlock(i);
     }
 
-    if (n_ready < 4) {
+    if (sids_in_track.len < 4) {
```

Comment:
Do I miss a point or the code still counts tracking channels instead of satellites?